sql/core/src/test/resources/sql-tests/analyzer-results/null-handling.sql.out

```diff
@@ -69,6 +69,24 @@ Project [a#x, (b#x + c#x) AS (b + c)#x]
    +- Relation spark_catalog.default.t1[a#x,b#x,c#x] parquet
 
 
+-- !query
+select b + 0 from t1 where a = 5
```

Comment:
I think we should test `b * 0`, where the optimization should be skipped to respect the NULL semantic.

Reply:
There are already such test cases in [ReorderAssociativeOperatorSuite.scala](https://github.com/apache/spark/pull/48395/files/c85d125a2e184c1c770bcba86dbc8d645f8d23eb#diff-16e48d6b354c9d83d4d4b4f56cefbd5c1d746062c0f050766008a28a2a8f93a3)

Reply:
hmm, then do we need to add additional tests in this golden file?

Reply:
`ReorderAssociativeOperatorSuite` is more about the plan correctness.

If we want the test to be more intuitive and specific for NULL semantics, I can add some more null-relevant cases here

Reply:
NVM, I guess we already have such test cases https://github.com/apache/spark/pull/48395/files#diff-c3c4e84c1d15e1c494ce6c062e78a9d53128a489d4831af6940bda50a1148d38R16-L15